airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -670,10 +680,16 @@ def write_logs(self, pod: k8s.V1Pod):
 
     def post_complete_action(self, *, pod, remote_pod, **kwargs):
         """Actions that must be done after operator finishes logic of the deferrable_execution."""
-        self.cleanup(
-            pod=pod,
-            remote_pod=remote_pod,
-        )
+        try:
+            self.cleanup(
+                pod=pod,
+                remote_pod=remote_pod,
+            )
+        except Exception:
+            # If one task got makred as failed, it should not raise exception which might cause retry.
+            # https://github.com/apache/airflow/issues/36471
+            if not self._killed:
+                raise
```

Comment:
isn't good idea to wrap the original exception with airflow exception and raise?

Reply:
Hmmm... I would say I don't really like wrapping the original exception as an airflow exception. I use airflow exception for new customized. exception

Reply:
agree, no need wrap